packages/integration-tests/test/actions.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { Config } from '../../../common'
+import { Web3Provider } from '@ethersproject/providers'
+import { ganache } from '@eth-optimism/ovm-toolchain'
+import assert = require('assert')
+
+const {
+  Contract,
+  ContractFactory,
+  providers: { JsonRpcProvider },
+  Wallet,
+} = require('ethers');
+const fs = require('fs')
+
+const sleep = ms => new Promise(resolve => setTimeout(resolve, ms))
+
+let SimpleStorage
+let L2Messenger
+const L1_USER_PRIVATE_KEY = Config.L1UserPrivateKey()
+const L2_USER_PRIVATE_KEY = Config.L2UserPrivateKey()
+const goerliURL = Config.L1NodeUrlWithPort()
+const optimismURL = Config.L2NodeUrlWithPort()
+const l1Provider = new JsonRpcProvider(goerliURL)
+const l2Provider = new JsonRpcProvider(optimismURL)
+const l1Wallet = new Wallet(L1_USER_PRIVATE_KEY, l1Provider)
+const l2Wallet = new Wallet(L2_USER_PRIVATE_KEY, l2Provider)
+const messengerJSON = JSON.parse(fs.readFileSync('contracts/build/iOVM_BaseCrossDomainMessenger.json'))
+const l2MessengerJSON = JSON.parse(fs.readFileSync('contracts/build/OVM_L2CrossDomainMessenger.json'))
+
+const deploySimpleStorage = async () => {
+  const SimpleStorageJson = JSON.parse(fs.readFileSync('contracts/build/SimpleStorage.json'))
+  const SimpleStorageFactory = new ContractFactory(SimpleStorageJson.abi, SimpleStorageJson.bytecode, l2Wallet)
+  SimpleStorage = await SimpleStorageFactory.deploy()
+}
+
+const deposit = async (amount, ...args) => {
+  const L1Messenger = new Contract(Config.L1MessengerAddress(), messengerJSON.abi, l1Wallet)
+  L2Messenger = new Contract(Config.L2MessengerAddress(), l2MessengerJSON.abi, l2Wallet)
```

Comment:
Why is `L2Messenger` a global?

Reply:
good catch. must have been an oversight. fixing!